pipelines/build/common/build_base_file.groovy

```diff
@@ -765,16 +765,33 @@ class Builder implements Serializable {
             tag = publishName
         }
 
+        if ((javaToBuild=="jdk21" || javaToBuild=="jdk") && scmReference && !release) {
```

Comment:
javaToBuild is eg.    jdk21u or jdk21 or jdk (for head)
javaVersion has no "u" and head is resolved to a number, eg. will be always "jdk21" or "jdk22"

so I think this line should be:
```
(javaVersion=="jdk21" || javaVersion=="jdk22")
```